src/audio_unit/looper.rs

```diff
@@ -6,13 +6,16 @@ use State::*;
 #[derive(Debug)]
 pub enum Message {
     Toggle,
+    EnableOverdubMode,
 }
 
 #[derive(Debug)]
 enum State {
     Off,
     Recording { position: usize },
     Playing { position: usize, total: usize },
+    PlayingAwaitingOverdub { position: usize, total: usize },
+    Overdubbing { position: usize, total: usize },
```

Comment:
3 of the cases having the same data makes me think we should pull that out into a struct that they all use.

Reply:
Maybe, this certainly isn't to the point of an abstraction "screaming at me" yet. For one thing, "nesting" `position` in three of them but leaving it at the top level of `Recording` feels like it disguises a symmetry

Reply:
Another thought here is that the `Playing` case could have an `awaitingOverdub` `bool` in it, rather than having a second `PlayingAwaitingOverdub` case.

Reply:
(That's not obviously better to me, I'm just throwing it out there.)

Reply:
Yeah, there's no way to give the default, but you could make `State::playing()` `State::playing_awaiting_overdub()` functions that do that for you. (Whether it's still worth putting those states together, I don't know.)